idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/symbols/KtFirSymbolProvider.kt

```diff
@@ -35,6 +35,9 @@ internal class KtFirSymbolProvider(
     private val firSymbolProvider by weakRef(firSymbolProvider)
 
     override fun getParameterSymbol(psi: KtParameter): KtValueParameterSymbol = withValidityAssertion {
+        if (psi.isFunctionTypeParameter) {
+            error("FirValueParameter for function type parameter is hidden inside the delegated type ref of the resolved type ref.")
```

Comment:
The error message should be about behaviour, not about the implementation. So, smth like "Creating KtValueParameterSymbol for functional type parameter is not possible. Please, see the KDoc of getParameterSymbol"

Reply:
Also, please consider describing this behaviour to the KDoc of corresponding method in `KtSymbolProviderMixin`

Reply:
Updated the error message as per the suggestion. Also added KDoc.